src/manage/getRequestFromRedirectUrl.ts

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright 2022 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { UrlString } from "@inrupt/solid-client";
+import {
+  isVerifiableCredential,
+  VerifiableCredential,
+} from "@inrupt/solid-client-vc";
+import {
+  REDIRECT_URL_PARAM_NAME,
+  REQUEST_VC_PARAM_NAME,
+  REQUEST_VC_URL_PARAM_NAME,
+} from "../discover/redirectToAccessManagementUi";
+import { getSessionFetch } from "../util/getSessionFetch";
+
+/**
+ * Get the Access Grant out of the incoming redirect from the Access Management app.
+ *
+ * @param redirectUrl The URL the user has been redirected to from the access
+ * management app.
+ * @param option Optional properties to customise the behaviour:
+ * - fetch: an authenticated fetch function. If not provided, the default session
+ * from @inrupt/solid-client-authn-browser will be used if available.
+ * @returns An Access Grant
+ */
+export async function getRequestFromRedirectUrl(
+  redirectUrl: UrlString,
+  option: { fetch?: typeof fetch } = {}
+): Promise<{ accessRequest: VerifiableCredential; redirectUrl: UrlString }> {
```

Comment:
```suggestion
): Promise<{ accessRequest: VerifiableCredential; accessRequestorRedirectUrl: UrlString }> {
```

The reason for which the redirectUrl return value is included here is a bit unclear.

It would be good to explain in the comment for the function...

Also, ultimately, I would probably rather provide a function to extract search parameters and standardise the function to retrieve a VC (which could become an extremely thin wrapper around 1 standard getVerifiableCredential function taking in a request param object which can be reused).

Reply:
This is quite similar to the OIDC flow: the requestor redirects the user to their access management app, and doing so they specify where the access management app should redirect the user back when access has been granted or denied. Agreed on factoring the function to get a VC out.

Reply:
Renamed in 45b6326446b33707ea2c197f47aedfd4da19cc91